openedx/core/djangoapps/util/user_messages.py

```diff
@@ -67,71 +67,115 @@ def css_class(self):
     def icon_class(self):
         """
         Returns the CSS icon class representing the message type.
-        Returns:
         """
         return ICON_CLASSES[self.type]
 
 
-def register_user_message(request, message_type, message, title=None):
-    """
-    Register a message to be shown to the user in the next page.
-    """
-    assert isinstance(message_type, UserMessageType)
-    messages.add_message(request, message_type.value, Text(message), extra_tags=EDX_USER_MESSAGE_TAG)
-
-
-def register_info_message(request, message, **kwargs):
-    """
-    Registers an information message to be shown to the user.
+class UserMessageCollection():
     """
-    register_user_message(request, UserMessageType.INFO, message, **kwargs)
-
-
-def register_success_message(request, message, **kwargs):
+    A collection of messages to be shown to a user.
     """
-    Registers a success message to be shown to the user.
-    """
-    register_user_message(request, UserMessageType.SUCCESS, message, **kwargs)
-
-
-def register_warning_message(request, message, **kwargs):
-    """
-    Registers a warning message to be shown to the user.
-    """
-    register_user_message(request, UserMessageType.WARNING, message, **kwargs)
+    @classmethod
+    @abstractmethod
+    def get_name(self):
+        """
+        Returns the name of the message collection.
 
+        The name is used to namespace the subset of django messages.
+        For example, return 'course_home_messages'.
+        """
+        raise NotImplementedError('Subclasses must define a namespace for messages.')
+
+    @classmethod
+    def get_message_html(self, body_html, title=None):
```

Comment:
Part of the issue is that this is missing doc-strings.
1. I imagine that most of the time there will be a 1:1 relationship between a MessageCollection class and some template.
2. This is the default for all collections, but any collection is free to override this for a different default for the message in its collection.
3. If certain messages in a collection need to be handled differently, they simply wouldn't supply the title and can do whatever they need in the body_html.
I'm not sure the best way to manage this in terms of code vs template, but I wanted to at least clarify the above.  Note that the rest of the message body html is in code and not in a template.
Does this change your feelings at all @andy-armstrong?